mutable_batch_pb/src/lib.rs

```diff
@@ -0,0 +1,358 @@
+//! Code to convert between binary write format and [`MutableBatch`]
+
+#![deny(rustdoc::broken_intra_doc_links, rustdoc::bare_urls, rust_2018_idioms)]
+#![warn(
+    missing_copy_implementations,
+    missing_debug_implementations,
+    missing_docs,
+    clippy::explicit_iter_loop,
+    clippy::future_not_send,
+    clippy::use_self,
+    clippy::clone_on_ref_ptr
+)]
+
+use hashbrown::HashSet;
+use snafu::Snafu;
+
+use generated_types::influxdata::pbdata::v1::{
+    column::{SemanticType, Values as PbValues},
+    Column as PbColumn, TableBatch,
+};
+use mutable_batch::{writer::Writer, MutableBatch};
+use schema::{InfluxColumnType, InfluxFieldType, TIME_COLUMN_NAME};
+
+/// Error type for line protocol conversion
+#[derive(Debug, Snafu)]
+#[allow(missing_docs)]
+pub enum Error {
+    #[snafu(display("error writing column {}: {}", column, source))]
+    Write {
+        source: mutable_batch::writer::Error,
+        column: String,
+    },
+
+    #[snafu(display("duplicate column name"))]
+    DuplicateColumnName { column: String },
```

Comment:
Would it be useful to have the column name that was duplicated in the error message?

```suggestion
    #[snafu(display("duplicate column name: {}", column))]
    DuplicateColumnName { column: String },
```